vm/actor/tests/init_actor_test.rs

```diff
@@ -0,0 +1,332 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod common;
+
+use actor::{
+    init::{ConstructorParams, ExecParams, ExecReturn, State},
+    Multimap, ACCOUNT_ACTOR_CODE_ID, FIRST_NON_SINGLETON_ADDR, INIT_ACTOR_CODE_ID,
+    MINER_ACTOR_CODE_ID, MULTISIG_ACTOR_CODE_ID, PAYCH_ACTOR_CODE_ID, POWER_ACTOR_CODE_ID,
+    STORAGE_POWER_ACTOR_ADDR, SYSTEM_ACTOR_ADDR, SYSTEM_ACTOR_CODE_ID,
+};
+use address::Address;
+use cid::Cid;
+use common::*;
+use db::MemoryDB;
+use ipld_blockstore::BlockStore;
+use message::{Message, UnsignedMessage};
+use serde::Serialize;
+use vm::{ActorError, ExitCode, Serialized, TokenAmount, METHOD_CONSTRUCTOR};
+
+fn construct_runtime<BS: BlockStore>(bs: &BS) -> MockRuntime<'_, BS> {
+    let receiver = Address::new_id(1000);
+
+    let message = UnsignedMessage::builder()
+        .to(receiver.clone())
+        .from(SYSTEM_ACTOR_ADDR.clone())
+        .build()
+        .unwrap();
+    let mut rt = MockRuntime::new(bs, message);
+    rt.caller_type = SYSTEM_ACTOR_CODE_ID.clone();
+    return rt;
+}
+
+// Test to make sure we abort actors that can not call the exec function
+#[test]
+fn abort_cant_call_exec() {
+    let bs = MemoryDB::default();
+    let mut rt = construct_runtime(&bs);
+    construct_and_verify(&mut rt);
+    let anne = Address::new_id(1001);
+
+    rt.set_caller(ACCOUNT_ACTOR_CODE_ID.clone(), anne);
+
+    let err = exec_and_verify(&mut rt, POWER_ACTOR_CODE_ID.clone(), &"")
+        .expect_err("Exec should have failed");
+    assert_eq!(err.exit_code(), ExitCode::ErrForbidden);
```

Comment:
The specs actor test execs and verifies on a cid.Undef as well. I think youre missing that

Reply:
can you give him the more specific details? Is it an option in the mockrt? which field specifically should be checked?

Reply:
Wouldn't a cid.Undef just be a invalid cid? Wouldn't we just return a error in that case?

Reply:
Youre right actually. I dont think we need to test for it. Sorry

Reply:
I think @dutterbutter  and I talked about it a while ago. He believed it was a non-issue